### PR TITLE
Fixing uncaught exception on window close

### LINKED ISSUE #10274: Uncaughtexception - Cannot read property 'close' of null at rpc-server.js:463
* Electron version: 1.7.3
* Operating system: Windows

### Expected behavior
No uncaught exception

### Actual behavior
Sometimes closing of new browserwindow hits this error

Cannot read property 'close' of null at EventEmitter.<anonymous> (rpc-server.js:463:39) 
at emitOne (events.js:96:13) 
at EventEmitter.emit (events.js:191:7) 
at WebContents.<anonymous> (browser\api\web-contents.js:256:13) 
at emitTwo (events.js:106:13) 
at WebContents.emit (events.js:194:7)

### How to reproduce
No consistent repro steps.

### PR DESCRIPTION
**Pull request by juturu**
Fixes #10274

### CODE REVIEW

File: lib/browser/rpc-server.js [hunk 1]
```diff
@@ -465,6 +465,10 @@ ipcMain.on('ELECTRON_BROWSER_WINDOW_CONFIRM', function (event, message, title) {
 
 // Implements window.close()
 ipcMain.on('ELECTRON_BROWSER_WINDOW_CLOSE', function (event) {
-  event.sender.getOwnerBrowserWindow().close()
```
Comment (MarshallOfSound) [inline lib/browser/rpc-server.js]: `try {} catch (err) {}` is quite costly if I recall correctly.

Can we just check if the window is `null` or not?  I.e.

```js
const window = event.sender.getOwnerBrowserWindow();
if (window) {
  window.close()
}
event.returnValue = null
```

Comment (MarshallOfSound) [inline lib/browser/rpc-server.js]: Or is the purpose of this to propagate the error rather than prevent it

Comment (juturu) [inline lib/browser/rpc-server.js]: I tried to follow how other methods in this file had try catch. I started with null check and changed into this based on other method implementation. Let me know if you feel we should change into this.

Comment (MarshallOfSound) [inline lib/browser/rpc-server.js]: I feel as though this should be a null check, looking through the rest of the file the other `try/catch` blocks are for potentially user influenced code.  Like `require` or calling functions or `global`.

This is just an edge case we need to handle I think with a null check (bug in code vs error that should be propagated).  Thoughts?

Comment (juturu) [inline lib/browser/rpc-server.js]: Makes sense. Updated, thanks @MarshallOfSound

Comment (YurySolovyov) [inline lib/browser/rpc-server.js]: I think try/catch is not a problem with TurboFan since Chrome 59

**Pull request merged**